controllers/api.js

```diff
@@ -1443,3 +1443,282 @@ exports.getTrakt = async (req, res, next) => {
+        }),
+
+      fetch(`https://pubchem.ncbi.nlm.nih.gov/rest/pug_view/data/compound/${aspirinCID}/JSON?heading=Density`)
+        .then((res) => res.json())
+        .catch((err) => {
+          console.error('Density API error:', err);
+          return { error: 'Failed to fetch density' };
+        }),
+
+      fetch(`https://pubchem.ncbi.nlm.nih.gov/rest/pug_view/data/compound/${aspirinCID}/JSON?heading=Vapor%20Pressure`)
+        .then((res) => res.json())
+        .catch((err) => {
+          console.error('Vapor Pressure API error:', err);
+          return { error: 'Failed to fetch vapor pressure' };
+        }),
+    ]);
+
+    // Search for similar compounds
+    const searchResults = await fetch(`https://pubchem.ncbi.nlm.nih.gov/rest/pug/compound/name/aspirin/cids/JSON?name_type=word`).then((res) => res.json());
+
+    // Get details for first 5 search results
+    let similarCompounds = [];
+    if (searchResults.IdentifierList && searchResults.IdentifierList.CID) {
+      const cids = searchResults.IdentifierList.CID.slice(0, 5);
+      const compoundDetails = await Promise.all(
+        cids.map((cid) =>
+          fetch(`https://pubchem.ncbi.nlm.nih.gov/rest/pug/compound/cid/${cid}/JSON`)
+            .then((res) => res.json())
+            .catch(() => null),
+        ),
+      );
+      similarCompounds = compoundDetails.filter(Boolean);
+    }
+
+    // Process and structure the data
+    const compound = compoundData.PC_Compounds?.[0] || {};
+    const properties = propertiesData.PropertyTable?.Properties?.[0] || {};
+
+    // Handle synonyms from API data
+    const synonyms = synonymsData.InformationList?.Information?.[0]?.Synonym || [];
+
+    const classifications = classificationData.Hierarchies || [];
+
+    // Extract safety information
+    const safetyInfo = {};
+    if (safetyData.Record?.Section) {
+      const safetySection = safetyData.Record.Section.find((s) => s.TOCHeading === 'Safety and Hazards');
+      if (safetySection?.Section) {
+        safetySection.Section.forEach((section) => {
+          if (section.TOCHeading && section.Information) {
+            safetyInfo[section.TOCHeading] = section.Information.map((info) => info.Value?.StringWithMarkup?.[0]?.String || info.Value?.String || '').filter(Boolean);
+          }
+        });
+      }
+    }
+
+    // Helper function to extract experimental property data
+    const extractExperimentalProperty = (data, propertyName) => {
+      console.log(`Processing ${propertyName}:`, JSON.stringify(data, null, 2));
+
+      if (data.error) {
+        console.log(`${propertyName} has error:`, data.error);
+        return { error: data.error };
+      }
+
+      const record = data.Record;
+      if (!record) {
+        console.log(`${propertyName} - No record found`);
+        return { values: [] };
+      }
+
+      const values = [];
+
+      // Function to recursively search through sections and references
+      const searchSections = (sections) => {
```

Comment:
Can we do what we need without a search function?  If we need to reduce the scope to drop this function, what do we need to drop?

Reply:
Have removed the above section as well, to reduce the size of the code